account_payment_term_extension/models/account_payment_term.py

```diff
@@ -173,10 +190,10 @@ def apply_payment_days(self, line, date):
         return date
 
     def compute(self, value, date_ref=False, currency=None):
-        """Complete overwrite of compute method to add rounding on line
-        computing and also to handle weeks and months
-        """
         self.ensure_one()
+        last_stock_move = False
+        if "last_stock_move" in self.env.context:
+            last_stock_move = self.env.context.get("last_stock_move")
```

Comment:
This could be simplified:
```suggestion
        last_stock_move = self.env.context.get("last_stock_move", False)
```